### ISSUE TITLE
[BUG]Running Dynamics

### ISSUE THREAD

**Issue by Meng-ho** [cagnulein/qdomyos-zwift#2291]
Hi Roberto,
Need your assistance. Thanks

(1) Screenshots

[debug-Fri_Apr_12_21_36_55_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14960274/debug-Fri_Apr_12_21_36_55_2024.log)

![Screenshot_20240412_222217_Drive](https://github.com/cagnulein/qdomyos-zwift/assets/166840338/f32b0826-0d10-40ea-a426-c8de110574c4)

(2) Description: Running metrics (Ground Contact Time and Vertical Oscillation) are NOT shown. I am only able to see the Stride length metric. Please refer to screenshot 2

(3) Equipment used
Spirit CT850 (FTMS treadmill) 
Garmin HRM Pro Plus ( with Running Dynamics)
Samsung S9 Ultra Tablet(Android) to run QZ

(4) QZ setting ( as advised by Roberto )
Power sensor options: 
Power sensor as Treadmill DISABLED.
Power sensor: Garmin HRM Pro Plus

Cadence sensor options:
Cadence sensor DISABLED

(5)
[debug-Fri_Apr_12_22_19_10_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14960507/debug-Fri_Apr_12_22_19_10_2024.log)
Desired outcome: To be able to see Running metrics for 
Ground Contact Time and Vertical Oscillation.

**Comment by Meng-ho** [comment]
Thanks Roberto

**Comment by cagnulein** [comment]
hi @Meng-ho are you sure your HRM pro device supports these metrics on bluetooth? I'm sure it's sending them over ANT+ but i'm not sure about bluetooth. Are you able to get this information from its manual? Let me know

**Comment by Meng-ho** [comment]
Hi Roberto,
Below are the specs for Garmin HRM Pro Plus

Physical Specifications

   - Unit dimensions (LxWxD): 23.5”-56.0” x 1.2” x 0.5” (60.0-142.0 x 3.0 x
   1.2 cm)
   - Weight: 1.8 oz (52 g)
   - Module size: 29.6 x 53.7 x 8.6 mm
   - Size adjustment: Bi-fold
   - Sizing range: 23.5”-42.0” (60.0-106.0 cm); 23.5”-56.0” (60.0-142.0 cm)
   with optional strap extender
   - Battery: CR2032
   - Battery door: Tool-free
   - Battery life: 12 months (Tri training 1 hour per day)
   - Water rating: 5 ATM
   - Operating temperature: 32°F-104°F (0°C-40°C)
   - Radio frequency/protocol: 2.4 GHz ANT wireless communications
   protocol; BLUETOOTH® Wireless Technology 5.0
   - Number of concurrent BLE connections supported: 3
   - Number of concurrent ANT connections supported: No limit
   - Range: 3 m
   - System Compatibility: ANT+, BLUETOOTH

Additional

   - ANT+® heart rate: yes
   - BLE heart rate: yes
   - Supports heart rate variability and advanced heart rate features: yes
   - ANT+ running dynamics: yes
   - Stand-alone activity monitoring (steps, all-day heart rate, calories
   and intensity minutes)
   - Supports TrueUp
   - Stores and forwards heart rate: yes
   - Swim interval heart rate statistics: yes
   - Running pace and distance: yes

Regards
Meng

**Comment by cagnulein** [comment]
yes as I thought the issue is there

`   - ANT+ running dynamics: yes`

this means that the only way to get those are by ANT+

Do you have a phone with ANT+ or do you have a ANT+ dongle?

**Comment by Meng-ho** [comment]
Yes, I am currently using ANT+ dongle on my Samsung S9 Ultra (Android)
tablet.

**Comment by cagnulein** [comment]
Ok I will investigate a way to collect them! Are you already able to get these metrics from an ANT+ app from the play store?

**Comment by Meng-ho** [comment]
No, I am not able to get these metrics  from an ANT+ app from the play
store. I hope QZ will allow me to get these metrics. 😀

**Comment by cagnulein** [comment]
i will check it and in case i will create a new branch for this

**Comment by cagnulein** [comment]
[D00001679_ANT+_Device_Profile_-_Running_Dynamics_Rev_1.1_(2) (1).pdf](https://github.com/cagnulein/qdomyos-zwift/files/14977043/D00001679_ANT%2B_Device_Profile_-_Running_Dynamics_Rev_1.1_.2.1.pdf)
this is the spec file

**Comment by Meng-ho** [comment]
Yes, my plug in Ant+ is working. I cannot have the HR metric shown in QZ when I disable HRM Pro plus from heart and power sensor, and use Ant+ Heart

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 7:33 pm, Roberto ***@***.***> wrote:   

so is it working with any play store app? it must be

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 15 apr 2024 alle 13:32 Meng-ho ***@***.***> ha
scritto:

> I am not getting HR metric from Ant+.
>
>
>
>
> Yahoo Mail: Search, organise, conquer
>

**Comment by cagnulein** [comment]
create a bug report and i will check it
https://developer.android.com/studio/debug/bug-report

this is the first step to get all the metrics


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 15 apr 2024 alle 13:39 Meng-ho ***@***.***> ha
scritto:

> Yes, my plug in Ant+ is working. I cannot have the HR metric shown in QZ
> when I disable HRM Pro plus from heart and power sensor, and use Ant+ Heart
>
> Yahoo Mail: Search, organise, conquer
>

**Comment by Meng-ho** [comment]
Noted Roberto. I will create the bug report. Thanks

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 7:41 pm, Roberto ***@***.***> wrote:   

create a bug report and i will check it
https://developer.android.com/studio/debug/bug-report

this is the first step to get all the metrics


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 15 apr 2024 alle 13:39 Meng-ho ***@***.***> ha
scritto:

> Yes, my plug in Ant+ is working. I cannot have the HR metric shown in QZ
> when I disable HRM Pro plus from heart and power sensor, and use Ant+ Heart
>
> Yahoo Mail: Search, organise, conquer
>

**Comment by Meng-ho** [comment]
Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 8:58 pm, Orca ***@***.***> wrote:   Hi Roberto,I have just emailed you the bug report as requested. Please let me know whether these are the information you need.
On Mon, Apr 15, 2024, 7:45 PM ***@***.*** ***@***.***> wrote:

Noted Roberto. I will create the bug report. Thanks

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 7:41 pm, Roberto ***@***.***> wrote:   

create a bug report and i will check it
https://developer.android.com/studio/debug/bug-report

this is the first step to get all the metrics


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 15 apr 2024 alle 13:39 Meng-ho ***@***.***> ha
scritto:

> Yes, my plug in Ant+ is working. I cannot have the HR metric shown in QZ
> when I disable HRM Pro plus from heart and power sensor, and use Ant+ Heart
>
> Yahoo Mail: Search, organise, conquer
>

**Comment by cagnulein** [comment]
hi @Meng-ho i checked the log but i don't see qz running on it. Was it running with the ant heart setting enabled and connected to your fitness device?

**Comment by Meng-ho** [comment]
Sorry about that. I have just email you with QZ running connected to my Garmin HRM Pro and FTM treadmill

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 9:10 pm, Roberto ***@***.***> wrote:   


hi @Meng-ho i checked the log but i don't see qz running on it. Was it running with the ant heart setting enabled and connected to your fitness device?

**Comment by cagnulein** [comment]
got it

`04-15 21:17:56.626 10401  1977  2417 E HeartChannelController: No Device Found`

that's the issue.

in some other play store apps, the HRM is shown like an HEART DEVICE or a Running Dynamic one? Because I'm looking actually for an HR one

**Comment by Meng-ho** [comment]
Please let me know what other information you need to resolve the issue. 😀

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 9:34 pm, Roberto ***@***.***> wrote:   


got it

04-15 21:17:56.626 10401 1977 2417 E HeartChannelController: No Device Found

that's the issue.

in some other play store apps, the HRM is shown like an HEART DEVICE or a Running Dynamic one? Because I'm looking actually for an HR one

**Comment by cagnulein** [comment]
i will test tomorrow with an Ant Heart rate device that I have and I will check if i can get it. So i can assist you better. I will let you know!

**Comment by Meng-ho** [comment]
Noted. As always, thanks much Roberto for your top-notch support and the awesome QZ app.

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 15 Apr 2024 at 10:20 pm, Roberto ***@***.***> wrote:   


i will test tomorrow with an Ant Heart rate device that I have and I will check if i can get it. So i can assist you better. I will let you know!

**Comment by cagnulein** [comment]
@Meng-ho ok do these steps

https://github.com/cagnulein/qdomyos-zwift/assets/48216/04b14670-a480-4498-993f-1fc55dc5d7eb

And then run again qz. You should see the hr (at least the avg hr and the max).

If so collect another bug report

Let me know!

**Comment by Meng-ho** [comment]
Hi Roberto
I still not able to see the HR metric. I have downloaded Ant plugin
sampler. I have sent you the bug report

**Comment by cagnulein** [comment]
but did you see it in the sampler as i did?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 16 apr 2024 alle 13:25 Meng-ho ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
that's so strange: I had the same your issue this morning and I fixed it in that way. Try, in the plugin sampler to press the "multi device search plugin", select all and press start search. it should find the hr press it and then close it again. then try again in qz (qz must be closed during this process). I will test this also with another HR too

**Comment by cagnulein** [comment]
if you want you can also test this build https://github.com/cagnulein/qdomyos-zwift/actions/runs/8704029422/artifacts/1417812609

Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.
Let me know if you have any questions

**Comment by Meng-ho** [comment]
I have downloaded the test build link. Do I need to do the setting as what
we did previously.

**Comment by cagnulein** [comment]
yes just ant heart

**Comment by Meng-ho** [comment]
Sorry Roberto. Am I suppose to try the test build 1 hour later. I have just
tested it. I will email you for the extension of the trial time to try 1
hour later

**Comment by cagnulein** [comment]
you can run it as many as you want. the 15 minutes is for session. anyway yah sure I can unlock it for you.

**Comment by Meng-ho** [comment]
Thanks. I will do the test build trial 1hour later. As of now testing, no
HR and Running metric shown on QZ. Good news is that I saw HRMPro+ detected
when I start the QZ now. 😀

**Comment by cagnulein** [comment]
ok in case send me again a bug report. it seems like your hrm is not sending the hr in broadcast mode.

**Comment by Meng-ho** [comment]
Ok. I will send you the bug report 45 minutes later as I need to wait for 1
hour for the test build before I test again.

**Comment by Meng-ho** [comment]
I have just email.you the bug report. From the testing of trial build, HR metric is still not shown on QZ. HRMpro+ is detected when starting QZ. 

Yahoo Mail: Search, organise, conquer 
 
  On Tue, 16 Apr 2024 at 8:56 pm, Orca ***@***.***> wrote:   Ok. I will send you the bug report 45 minutes later as I need to wait for 1 hour for the test build before I test again. 

On Tue, Apr 16, 2024, 8:49 PM Roberto Viola ***@***.***> wrote:




ok in case send me again a bug report. it seems like your hrm is not sending the hr in broadcast mode.

**Comment by cagnulein** [comment]
i don't understand which is the difference from my setup and yours. i can read HR from 2 different ANT heart devices with this build. Are you able to record a video from the ant plugin sampler to qz so i will understand if you are doing something wrong?

**Comment by Meng-ho** [comment]
I have just email video and also screenshots of my settings on Samsung Device. Let me know what are infor you need. Btw, what additional steps I need yo do after the hrm is shown in ant plugin sampler 

Yahoo Mail: Search, organise, conquer 
 
  On Tue, 16 Apr 2024 at 10:32 pm, Roberto ***@***.***> wrote:   


i don't understand which is the difference from my setup and yours. i can read HR from 2 different ANT heart devices with this build. Are you able to record a video from the ant plugin sampler to qz so i will understand if you are doing something wrong?

**Comment by cagnulein** [comment]
when you see the hrm on the plugin sampler, i just swipe out this app and i open qz, i'm seeing the hr when i do this

**Comment by Meng-ho** [comment]
I have just tried and still.not able to see the HR metric. It is probably due to my HRM device. 
I am happy just to go back to using bluetooth for HR metric and ok not to have the rumning metric. The QZ app is still awesome without the running metric and I thank you for your patience in resolving the issue.
Thanks again Roberto. 



Yahoo Mail: Search, organise, conquer 
 
  On Wed, 17 Apr 2024 at 1:19 am, Roberto ***@***.***> wrote:   


when you see the hrm on the plugin sampler, i just swipe out this app and i open qz, i'm seeing the hr when i do this

**Comment by cagnulein** [comment]
i will think about this. i guess i have to implement somehow a direct ant+
connection to it


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 16 apr 2024 alle 19:51 Meng-ho ***@***.***> ha
scritto:

> I have just tried and still.not able to see the HR metric. It is probably
> due to my HRM device.
> I am happy just to go back to using bluetooth for HR metric and ok not to
> have the rumning metric. The QZ app is still awesome without the running
> metric and I thank you for your patience in resolving the issue.
> Thanks again Roberto.
>
>
>
> Yahoo Mail: Search, organise, conquer
>

**Comment by Meng-ho** [comment]
Noted. Thanks Roberto. As for now, I will continue to use bluetooth for my HR metric (without Running Dynamics) and enjoy the awesome QZ for my run.
 
Yahoo Mail: Search, organise, conquer 
 
  On Wed, 17 Apr 2024 at 2:35 am, Roberto ***@***.***> wrote:   

i will think about this. i guess i have to implement somehow a direct ant+
connection to it


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 16 apr 2024 alle 19:51 Meng-ho ***@***.***> ha
scritto:

> I have just tried and still.not able to see the HR metric. It is probably
> due to my HRM device.
> I am happy just to go back to using bluetooth for HR metric and ok not to
> have the rumning metric. The QZ app is still awesome without the running
> metric and I thank you for your patience in resolving the issue.
> Thanks again Roberto.
>
>
>
> Yahoo Mail: Search, organise, conquer
>
